### ISSUE TITLE
Accessibility on Search Dialog

### ISSUE THREAD

**Issue by VinoManoTR** [scalar/scalar#4058]
### What happens?

1. The screen reader doesn't announce the search dialog when it is opened.
2. Once the search dialog is closed, the focus doesn't return to the search bar that opened the dialog.

3. Search dialog doesn't announce the number of results or if no results are found.
![image](https://github.com/user-attachments/assets/71dbc0ee-3734-4350-946d-a0f48b4839db)

4. Footer labels are not announced by screen reader.
![image](https://github.com/user-attachments/assets/b026745d-d369-4c74-a2ba-1467f5c2b65b)

5. On Screen resolution is 1280x1024 and zoom the page to 300% all content needs to be visible/accessible but currently the search bar and footer is visible but the first entry is cut off. We should be able to see at least one option of content.
![image](https://github.com/user-attachments/assets/a76c9ffb-c6c0-448c-aad6-2976b1fbc662)

6. We should have a “X” close button and be able to close with “ESC” button .

### What did you expect to happen?

All active elements on the test request page should be keyboard accessible.
When the keyboard focus is on an element, there should be a visual indicator of the focus (usually a box highlighting the element).
The screen reader should be able to make an announcement of each element that includes context for the visual label.

**Comment by hwkr** [comment]
Hi @VinoManoTR! Just digging into this, I'm having trouble reproducing some of the issues so I'll probably need you to upload a screen recording of the issue (or similar) along with the browsers, platforms and screen readers you've tested with.

Here's a quick breakdown of my investigation:

1. The screen reader doesn't announce the search dialog when it is opened. - It should, the form is hosted inside a dialog and focus is pushed to the first input according to the [ARIA dialog pattern](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/#keyboard_interaction). The first input is a search input with following an [ARIA combobox pattern](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/), I had a look around and this was the pattern that seemed to best align with our use case but if there's a better pattern available I'd be happy to adopt it. 
2. Once the search dialog is closed, the focus doesn't return to the search bar that opened the dialog - It should, this could be browser dependent though and also screen readers (e.g. voiceover) sometimes intercept the browser focus.
3. Search dialog doesn't announce the number of results or if no results are found. - Typically the screen readers will count the number of items in a listbox but this is also screen reader dependent. We could add it to the listbox label but I'm hesitant to change things to cater to a particular screen reader rather than the WCAG standards.
4. Footer labels are not announced by screen reader. - They should be, the instructions are linked to the search input via an `aria-describedby`. Wether or not the label is read is screen reader dependent though.
5. On Screen resolution is 1280x1024 and zoom the page to 300% all content needs to be visible/accessible but currently the search bar and footer is visible but the first entry is cut off. We should be able to see at least one option of content. - It seems like we have some fixed values in our modal sizing that are preventing it from filling the viewport, I'll open a PR to fix it :)
6. We should have a “X” close button and be able to close with “ESC” button. - Pressing ESC should already close the modal and mouse / touch users can close also the modal by clicking outside the dialog. As far as I can tell the [ARIA dialog pattern](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/#keyboard_interaction) specifies closing the dialog with ESC but not including a specific close button. We could potentially however add a keyboard focusable control to close the dialog as well.

Let me know if you have any questions or if any of this doesn't make sense!
